assets/asset_manager.cc

```diff
@@ -29,6 +29,29 @@ void AssetManager::PushBack(std::unique_ptr<AssetResolver> resolver) {
   resolvers_.push_back(std::move(resolver));
 }
 
+void AssetManager::UpdateResolverByType(
+    std::unique_ptr<AssetResolver> updated_asset_resolver,
+    AssetResolver::AssetResolverType type) {
+  bool updated = false;
+  std::deque<std::unique_ptr<AssetResolver>> new_resolvers;
+  for (auto& old_resolver : resolvers_) {
+    if (!updated && old_resolver->GetType() == type) {
```

Comment:
This behavior is somewhat counterintuitive - if updated_asset_resolver is null, then all existing resolvers of the specified type will be removed.  But if updated_asset_resolver is non-null, then only the first resolver of that type will be replaced.

Is it necessary to allow updated_asset_resolver to be null?  If there is no use case for that, then I'd prefer to limit this API to just replacing the first resolver of the given type.

Reply:
Made nullptr input into a no-op.